### PR TITLE
merging the annotation testing into the matrix inversion branch

### PR DESCRIPTION
**Pull request by ocalex86**

### CODE REVIEW

File: Compiler/types.py [hunk 1]
```diff
@@ -1774,6 +1774,10 @@ class sfloat(_number):
     kappa = 40
     round_nearest = False
     error = 0
+    reg_type = 'sf'
+
+    def conv(self):
```
Comment (ocalex86) [inline Compiler/types.py]: what is this for? Do we really need it?

File: Compiler/types.py [hunk 2]
```diff
@@ -2316,12 +2332,15 @@ def __repr__(self):
 
 
 class MemFloat(_mem):
+    __slots__ = ['reg_type']
```
Comment (ocalex86) [inline Compiler/types.py]: What' is this for?

File: Programs/Source/linreg.mpc [hunk 3]
```diff
@@ -13,64 +13,83 @@ def procData(yfname, xfname):
...
 
 # computes scalar product of two vectors of sfloats
 def scalarProd(v, w):
-    sum = sfloat(0)
-    for i in range(len(v)):
-        sum = sum + v[i]*w[i]
-    return sum
+    sum = Array(1,sfloat)
+    sum[0] = sfloat(0)
+    #TODO improve
```
Comment (djetchev) [inline Programs/Source/linreg.mpc]: remove TODO

File: Programs/Data/X.3.485.txt [hunk 4]
```diff
@@ -1,16 +1,16 @@
 1,6.1101,6.1101
```
Comment (ngama75) [inline Programs/Data/X.3.485.txt]: Do we need to have data in the git? 
if we keep the original one data file, does it still work?

File: Programs/Source/linreg.mpc [hunk 5]
```diff
@@ -13,64 +13,83 @@ def procData(yfname, xfname):
...
-    # extend
-    for i in xrange(k):
-        A[i] = A[i] + [sfloat(0)] * (i) + [sfloat(1)] + [sfloat(0)] * (k - i - 1)
-    # TODO test if 0
-    for p in xrange(k):
+def inverse(M):
+    length = len(M)
+    A = Matrix(length, 2*length, sfloat)
+    #Extend (write unit matrix)
+    for i in xrange(length):
```
Comment (djetchev) [inline Programs/Source/linreg.mpc]: maybe try annotated loops here

File: Programs/Source/exp.mpc [hunk 6]
```diff
@@ -1,9 +1,16 @@
...
+increment = Array(1, sfloat)
+increment[0] = base[0]
+exp = Array(1, sfloat)
+exp[0] = sfloat(1)
 iterations = 20
-for i in xrange(1,iterations):
-    exp = exp + increment/sfloat(math.factorial(i))
-    increment = base*increment
-print_ln("exp((%s,%s,%s,%s)) = (%s,%s,%s,%s)", base.v.reveal(), base.p.reveal(), base.z.reveal(), base.s.reveal(), exp.v.reveal(), exp.p.reveal(), exp.z.reveal(), exp.s.reveal())
+facts = Array(iterations, sfloat)
```
Comment (ngama75) [inline Programs/Source/exp.mpc]: maybe put a comment saying that facts is a constant table of factorials

File: Programs/Source/linreg.mpc [hunk 7]
```diff
@@ -13,64 +13,83 @@ def procData(yfname, xfname):
...
+            A[i][j] = M[i][j]
+            if j == i:
+                A[i][length+j] = sfloat(1)
+            else:
+                A[i][length+j] = sfloat(0)
+
+    #initial step
+    @for_range(length)
+    def f(p):
         pivot = A[p][p]
```
Comment (ngama75) [inline Programs/Source/linreg.mpc]: maybe, see if it is faster to store 1/pivot

File: Compiler/types.py (hunk 1 above)
Comment (ocalex86) [inline Compiler/types.py]: seems required

**Pull request merged**